pkg/catalog/routes.go

```diff
@@ -102,6 +102,38 @@ func (mc *MeshCatalog) ListAllowedOutboundServices(sourceService service.MeshSer
 	return mc.getAllowedDirectionalServices(sourceService, outbound)
 }
 
+// ListAllowedOutboundServicesForIdentity list the services the given service account is allowed outbound connections to
+func (mc *MeshCatalog) ListAllowedOutboundServicesForIdentity(identity service.K8sServiceAccount) []service.MeshService {
+	serviceSet := mapset.NewSet()
+	for _, t := range mc.meshSpec.ListTrafficTargets() { // loop through all traffic targets
+		for _, source := range t.Spec.Sources {
+			if source.Name == identity.Name && source.Namespace == identity.Namespace { // found outbound
```

Comment:
nit: inverting the condition to `if not () continue` will reduce indent due to nesting.